lib/ansible/modules/pip.py

```diff
@@ -611,6 +614,62 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _normalize_vcs_packages(
+        module: AnsibleModule,
+        package_list: list[Package],
+        pip: list[str]
+) -> list[Package]:
+    """Return a list of packages with names extracted from metadata in VCS"""
+    vcs_packages = [pkg_str for pkg in package_list if _is_vcs_url(pkg_str := str(pkg))]
+
+    if not vcs_packages:
+        return package_list
+
+    # Unsetting 'FORCE_COLOR' with 'NO_COLOR' and TTY_COMPATIBLE settings, helps to
+    # prevent color codes from being injected into and corrupting the JSON output.
+    os.environ.pop('FORCE_COLOR', None)
+
+    rc, json_out, err = module.run_command(
+        [*pip, 'install', '--dry-run', '--ignore-installed', '--quiet', '--report=-', *vcs_packages],
+        environ_update={
+            'NO_COLOR': '1',
+            'TTY_COMPATIBLE': '0',
+        },
+    )
+
+    if rc == 0:
+        # If it succeeds, handle the report
+        report = json.loads(json_out)
+        package_objects = (Package(install_report['metadata']['name'], version_string=install_report['metadata']['version'])
+                           for install_report in report['install'])
+    else:
+        # Else, if that fails due to --dry-run not being present in pip
+        # versions older than 22.2 (ex: Ubuntu 22.04), fall back to using
+        # the potentially faulty pip download method and warn the user.
+
+        pip_downloads_dir = Path(module.tmpdir) / 'pipdownloads'
```

Comment:
I recommend using kebab-case for file paths. This sets them apart from snake_case var names in code, making the difference more visual.
```suggestion
        pip_downloads_dir = Path(module.tmpdir) / 'pip-downloads'
```